File: SWEA/D2/2001. 파리 퇴치/파리 퇴치.py
t = int(input())

for tc in range(1, t+1):
    n, m = map(int, input().split())
    matrix = [list(map(int, input().split())) for _ in range(n)]
    # print(matrix)
    kill_log = []

    for i in range(n):
        for j in range(n):
            now = 0
            for m1 in range(m):
                for m2 in range(m):
                    if i+m1 < n and j + m2 < n:
                        now += matrix[i+m1][j+m2]
            kill_log.append(now)
    print(f'#{tc} {max(kill_log)}')

File: SWEA/D2/2001. 파리 퇴치/파리 퇴치.java
import java.util.Scanner;
public class Solution {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        int t = sc.nextInt();

        for (int tc = 1; tc <= t; tc++) {
            int n = sc.nextInt();
            int m = sc.nextInt();
            int[][] matrix = new int[n][n];
            for (int i = 0; i < n; i++) {
                for (int j = 0; j < n; j++) {
                    matrix[i][j] = sc.nextInt();
                }
            }
            int[] killLog = new int[n * n * m * m];
            int index = 0;

            for (int i = 0; i < n; i++) {
                for (int j = 0; j < n; j++) {
                    int now = 0;
                    for (int m1 = 0; m1 < m; m1++) {
                        for (int m2 = 0; m2 < m; m2++) {
                            if (i + m1 < n && j + m2 < n) {
                                now += matrix[i + m1][j + m2];
                            }
                        }
                    }
                    killLog[index++] = now;
                }
            }

            int maxKill = Integer.MIN_VALUE;
            for (int k : killLog) {
                maxKill = Math.max(maxKill, k);
            }
            System.out.println("#" + tc + " " + maxKill);
        }
    }
}